### PR TITLE
Descriptive names for ci runs

### PR DESCRIPTION
**Pull request by gusty1g**
## Description

Summary of changes:
- Added descriptive names to jobs in CI. Earlier, when the CI checks after the run was complete, it will show test, test, ... because multiple jobs had same names. The change adds descriptive names, allowing to easily catch failed jobs.
- Removed macOS installation and cache collection steps (mostly `if` conditions) since our CI does not run in macOS. Only the build CI runs in macOS and macOS installation is preserved there.
- Minor improvement to pull request template.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .github/workflows/jax_setup.yml [hunk 1]
```diff
@@ -62,30 +62,13 @@ jobs:
...
-    - name: Cache pip packages for Linux
-      if: runner.os == 'Linux'
+    - name: Cache pip packages
       uses: actions/cache@v3
       with:
         path: ~/.cache/pip
         key: ${{ runner.os }}-pip-${{ hashFiles('requirements/jax/**') }}
         restore-keys: |
           ${{ runner.os }}-pip-
-    - name: Cache pip packages for MacOS
```
Comment (rbharath) [inline .github/workflows/jax_setup.yml:73]: Pytest runs on mac right? Let's chat offline

**Review by rbharath — COMMENTED** [review]
There are multiple unrelated changes in this PR. I'd suggest splitting up into a few smaller PRs (pull request template simplification, descriptive names, cache changes). I'm not sure about the cache change in particular, so let's discuss that offline

**Comment by gusty1g** [comment]
> There are multiple unrelated changes in this PR. I'd suggest splitting up into a few smaller PRs (pull request template simplification, descriptive names, cache changes). I'm not sure about the cache change in particular, so let's discuss that offline

I have made cache changes into a separate pull request (#3342 )

**Review by rbharath — APPROVED** [review]
LGTM

Unrelated to this PR, it looks like there is a Grover failure:

```
_________________________ testGroverBondVocabPredictor _________________________

    @pytest.mark.torch
    def testGroverBondVocabPredictor():
        from deepchem.models.torch_models.grover_layers import GroverBondVocabPredictor
        num_bonds = 20
        in_features, vocab_size = 16, 10
        layer = GroverBondVocabPredictor(vocab_size, in_features)
        embedding = torch.randn(num_bonds * 2, in_features)
>       result = layer(embedding)

deepchem/models/torch_models/tests/test_grover_layers.py:49: 
```

Can you take a look and address in a follow up PR @gusty1g ?

**Comment by gusty1g** [comment]
The failure in grover was addressed in #3335. Since this branch is not updated with master branch, it is popping up here.

**Pull request merged**